aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -1061,194 +1062,152 @@ module aptos_framework::stake {
         });
 
         // Process pending stake and distribute transaction fees and rewards for each currently pending_inactive validator
         // (requested to leave but not removed yet).
         vector::for_each_ref(&validator_set.pending_inactive, |validator| {
             let validator: &ValidatorInfo = validator;
-            update_stake_pool(validator_perf, validator.addr, &config);
+            let new_stake = update_stake_pool(validator_perf, validator.addr, &config, commit);
+            simple_map::add(&mut new_stakes_by_validator, validator.addr, new_stake);
         });
 
-        // Activate currently pending_active validators.
-        append(&mut validator_set.active_validators, &mut validator_set.pending_active);
-
-        // Officially deactivate all pending_inactive validators. They will now no longer receive rewards.
-        validator_set.pending_inactive = vector::empty();
+        // Get the list of validators who intend to be in the next validator set.
+        let candidates = addresses_from_validator_infos(&validator_set.active_validators);
+        vector::reverse_append(&mut candidates, addresses_from_validator_infos(&validator_set.pending_active));
 
-        // Update active validator set so that network address/public key change takes effect.
-        // Moreover, recalculate the total voting power, and deactivate the validator whose
-        // voting power is less than the minimum required stake.
+        // Go through the candidate list, drop those without enough stakes, construct a new validator set and determine their indices.
         let next_epoch_validators = vector::empty();
+        let next_validator_idx = 0;
         let (minimum_stake, _) = staking_config::get_required_stake(&config);
-        let vlen = vector::length(&validator_set.active_validators);
+        let num_candidates = vector::length(&candidates);
         let total_voting_power = 0;
-        let i = 0;
+        let candidate_idx = 0;
         while ({
             spec {
                 invariant spec_validators_are_initialized(next_epoch_validators);
             };
-            i < vlen
+            candidate_idx < num_candidates
         }) {
-            let old_validator_info = vector::borrow(&mut validator_set.active_validators, i);
-            let pool_address = old_validator_info.addr;
-            let validator_config = borrow_global<ValidatorConfig>(pool_address);
-            let stake_pool = borrow_global<StakePool>(pool_address);
-            let new_validator_info = generate_validator_info(pool_address, stake_pool, *validator_config);
+            let pool_address = *vector::borrow(&mut candidates, candidate_idx);
+            let new_voting_power = if (simple_map::contains_key(&new_stakes_by_validator, &pool_address)) {
+                *simple_map::borrow(&new_stakes_by_validator, &pool_address)
+            } else {
+                let candidate_stake_pool = borrow_global<StakePool>(pool_address);
```

Comment:
This is only for pending_active validators right?

Reply:
correct.